app/lib/features/space/settings/pages/visibility_accessibility_page.dart

```diff
@@ -202,4 +266,45 @@ class _VisibilityAccessibilityPageState
       );
     }
   }
+
+  Future<void> updateSpaceVisibility(
+    RoomVisibility value,
+    String spaceId, {
+    String? roomId,
+  }) async {
+    try {
+      EasyLoading.show(status: 'Updating space settings', dismissOnTap: false);
+      final sdk = await ref.read(sdkProvider.future);
+      final room = await ref.read(maybeRoomProvider(spaceId).future);
+      final space = await ref.read(maybeSpaceProvider(spaceId).future);
+      final update = sdk.api.newJoinRuleBuilder();
+      switch (value) {
+        case RoomVisibility.Public:
+          update.joinRule('public');
+          break;
+        case RoomVisibility.Private:
+          update.joinRule('invite');
+          break;
+        case RoomVisibility.SpaceVisible:
+          update.joinRule('restricted');
+          if (roomId != null) {
+            if (!(await space!.isChildSpaceOf(roomId))) {
+              await room!.addParentRoom(roomId, true);
+              break;
+            } else {
+              await space.addChildRoom(roomId);
+              break;
```

Comment:
Same as above comment, limited permission is all about adding child and removing child?